### ISSUE TITLE
Deprecation warning for Django 4.0

### ISSUE THREAD

**Issue by larsrinn** [jazzband/django-two-factor-auth#381]
Since the upgrade to Django 3.1, I get some deprecation warnings for an upgrade to Django 4.0:

```
two_factor/signals.py:3: RemovedInDjango40Warning: The providing_args argument is deprecated. As it is purely documentational, it has no replacement. If you rely on this argument as documentation, you can move the text to a code comment or docstring.
    user_verified = Signal(providing_args=['request', 'user', 'device'])
```

and some like 

```
two_factor/urls.py:9
  /usr/local/lib/python3.8/site-packages/two_factor/urls.py:9: RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
```

## Expected Behavior

Not use any code that's marked for deprecation

## Current Behavior

Using code that's marked for deprecation

## Possible Solution

As the warning for the first issue states, it's safe to remove the `providing_args` argument.

For the second warning, I think it's sufficient to simply replace it with `django.urls.re_path`. But if the regex-bits are not required (I didn't look at the code), I'd suggest to migrate to the new patterns using `django.urls.path`.

## Steps to Reproduce (for bugs)

This were the bits that occured for me (using two factor auth with totp and static otps). Maybe there are some deprecated bits in the codebase that are not imported by my code?

## Context

Django 3.1 release notes: https://docs.djangoproject.com/en/3.1/releases/3.1/

`django.urls.path` has been added in 2.0: https://docs.djangoproject.com/en/2.0/ref/urls/#django.urls.re_path which is beyond it's end of life: https://www.djangoproject.com/download/#supported-versions , so I'd say it's fair for this package to drop support for anything that's older than 2.2 (the current LTS version) 

## Your Environment

* Browser and version: doesn't matter
* Python version: 3.8.5
* Django version: 3.1
* django-otp version: 0.9.4
* django-two-factor-auth version: 1.12.1
* Link to your project: Doesn't matter

**Comment by larsrinn** [comment]
I found one more:

```
/usr/local/lib/python3.8/site-packages/two_factor/views/core.py:158: RemovedInDjango40Warning:
  
  django.utils.http.is_safe_url() is deprecated in favor of url_has_allowed_host_and_scheme().
```

**Comment by moggers87** [comment]
Some of these have been fixed by #368 and some can't be resolved until 2.2 goes end-of-life.

**Comment by claudep** [comment]
I think all of them are fixed in the current master branch.
